content/Dev/remote_settings_history.rst

```diff
@@ -0,0 +1,179 @@
+
+The revolutionary ideals of our initial team were fading away. We had to admit that we failed at promoting our vision of a user-facing storage backend (see :ref:Thoughts section). Self hosting, decentralized Web, Tor hidden services, were conversations at night around a drink, not in meeting rooms anymore. I'm pretty sure Mozilla even had a few projects running on Google Firebase at this time.
+
+At the end of that period, most members of the original team had left, disillusioned. Ethan and I were the last «Kinto guys» at Mozilla.
+
+
+.. figure:: images/remote-settings-history-3-contributors.png
+
+	Fortunately we had dependabot enabled on our repos so we don't look too lonely.
+
+
+Modern Age (2018-2020)
+======================
+
+In december 2016, we made a list of all the sources where *Firefox* would pull data from. No spoiler, it was too long.
+
+Our goal was to unify several of these communication channels into one, and starting demanding a good reason to justify rolling out a new update source.
+
+For most use-cases in Firefox, the *Kinto* API was too low level, and doing too much. As an example, the certificate configuration for signature verification `was unnecessarily complex <https://searchfox.org/mozilla-esr60/rev/02b4ae79b24aae2346b1338e2bf095a571192061/services/common/blocklist-clients.js#439-474>`_. And there was no official cookbook.
+
+Our new manager Benson was defending the idea of *Remote Settings*. Two methods: ``.get()`` and ``.on("sync", ...)``.
+New collections of records had to be self-service, because we were not going to be able to continue the manual onboarding.
+
+We had all the pieces in place to achieve that, it was mostly about improving developer experience. We refactored a number of things, added OAuth support, and Ethan integrated our clients and server with the `broadcast service <https://github.com/mozilla-services/megaphone>`_ for push notifications. We morphed the existing client integrations into a high level generic API, available for any component in Firefox. And on the server side, the original Kinto Admin had everything in its DNA to become the *Remote Settings Admin UI*.
+
+.. image:: images/remote-settings-history-kinto-admin.png
+
+In December 2018, I presented *Remote Settings* in front of hundreds of Firefox developers! This felt energizing and great! On the other hand, I knew I was reaping the rewards of our team mates who had left.
+
+.. image:: images/remote-settings-history-sf-evergreen.png
+
+We merged with the *SHIELD* team and were now part of the *Product Delivery* team. Now responsible of making off-train changes reliable and safe (ie. updates without reinstall).
+
+Since we were only pulling read-only data from the server, for the sake of simplicity, I got rid of the *Kinto offline* library in Firefox and replaced it with plain IndexedDB code. Way easier to reason about, especially for a future code reader who wouldn't have the back story.
```

Comment:
Maybe mention that moving to implementation-specific code for ffx, the project lost most of its traction for generic OSS usage?

Reply:
pas compris....

*Since we were only pulling read-only data from the server, the whole offline bi-directionnal sync code was overkill and slightly inefficient.* ?

Reply:
Je voulais dire qu'en pilotant le dev de Kinto pour les seuls besoins de Firefox, on avait sans doute un peu perdu la vision "générique" que portait le projet. Mais en fait non pas forcément, donc oublie ^^